src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private double side;
+
+    public Square(String color, double side) {

```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. While `side` is not an expression but a single variable, for consistency with the task requirements, consider initializing `side` in the `FigureSupplier` and then passing it to the constructor.